tests/test_client.py

```diff
@@ -249,66 +211,295 @@ def test_get_crawl_results(client: Supadata, requests_mock) -> None:
 
 
 def test_gateway_error_429(client: Supadata, requests_mock) -> None:
-    '''Test handling of 429 gateway error.'''
+    """Test handling of 429 gateway error."""
     requests_mock.get(
-        f'{client.base_url}/youtube/transcript',
+        f"{client.base_url}/youtube/transcript",
         status_code=429,
-        text='You have exceeded the allowed request rate or quota limits'
+        text="You have exceeded the allowed request rate or quota limits",
     )
 
     with pytest.raises(SupadataError) as exc_info:
-        client.youtube.transcript(video_id='test123')
-    
+        client.youtube.transcript(video_id="test123")
+
+    error = exc_info.value
+    assert error.error == "limit-exceeded"
+    assert error.message == "Limit exceeded"
+    assert error.details == "You have exceeded the allowed request rate or quota limits"
+
+
+def test_youtube_video(client: Supadata, requests_mock) -> None:
+    video_id = "pEfrdAtAmqk"
+    mock_response = {
+        "id": video_id,
+        "duration": 1002,
+        "description": "The programming iceberg is complete roadmap to the loved, ...",
+        "title": "God-Tier Developer Roadmap",
+        "channel": {"id": "UCsBjURrPoezykLs9EqgamOA", "name": "Fireship"},
+        "tags": ["#iceberg", "#learntocode", "#programming"],
+        "thumbnail": "https://i.ytimg.com/vi/pEfrdAtAmqk/maxresdefault.jpg",
+        "uploadDate": "2022-08-24T00:00:00.000Z",
+        "viewCount": 7388353,
+        "likeCount": 262086,
+        "transcriptLanguages": ["en"],
+    }
+
+    requests_mock.get(
+        f"{client.base_url}/youtube/video?id={video_id}", json=mock_response
+    )
+
+    video = client.youtube.video(video_id)
+    assert isinstance(video, YoutubeVideo)
+    assert video.id == mock_response["id"]
+    assert video.duration == mock_response["duration"]
+    assert video.description == mock_response["description"]
+    assert video.title == mock_response["title"]
+    assert video.channel == mock_response["channel"]
+    assert video.tags == mock_response["tags"]
+    assert video.uploaded_date == datetime.fromisoformat(mock_response["uploadDate"])
+    assert video.view_count == mock_response["viewCount"]
+    assert video.like_count == mock_response["likeCount"]
+
+
+def test_youtube_video_invalid_id(client: Supadata, requests_mock) -> None:
+    video_id = "pEfrdAtmqk"
+    mock_response = {
+        "error": "not-found",
+        "message": "The requested item could not be found",
+    }
+    requests_mock.get(
+        f"{client.base_url}/youtube/video?id={video_id}",
+        status_code=404,
+        json=mock_response,
+    )
+
+    with pytest.raises(SupadataError) as exc_info:
+        client.youtube.video(video_id)
+
+    error = exc_info.value
+    assert error.error == "invalid-request"
+    assert error.message == "Endpoint does not exist"
+    assert error.details == "The requested item could not be found"
```

Comment:
is the assertion correct?

the error and message seem to be different than the mock response